File: 167-TwoSumIiInputArrayIsSorted/167-TwoSumIiInputArrayIsSorted.java
// Last updated: 6/10/2026, 7:26:10 PM
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l=0 , r = numbers.length-1;
        while (l < r) {
            int curr = numbers[l] + numbers[r];
            if(curr == target) return new int[]{l+1 , r+1};
            if(curr < target) l++;
            if(curr > target) r--; 
        }
        return new int[]{};
    }
}

File: 167-TwoSumIiInputArrayIsSorted/167-TwoSumIiInputArrayIsSorted.cpp
// Last updated: 6/10/2026, 7:26:14 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        int l=0 , r = n-1;
        while(l < r) {
            int sum = nums[l] + nums[r];
            if(sum < target)
                l++;
            else if(sum > target)
                r--;
            else {
                break;
            }
        }
        return {l+1 , r+1};
    }
};